app/src/sharedTest/java/org/oppia/app/activity/InputInteractionViewTestActivityTest.kt

```diff
@@ -89,4 +83,103 @@ class InputInteractionViewTestActivityTest {
+      assertThat(textAnswerRetriever.getPendingAnswer()).isInstanceOf(InteractionObject::class.java)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.denominator).isEqualTo(0)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.numerator).isEqualTo(0)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.wholeNumber).isEqualTo(0)
+    }
+  }
+
+  @Test
+  fun testFractionInputInteractionView_withInputtedText_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_fraction_input_interaction_view)).perform(typeText("9/10"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_fraction_input_interaction_view) as FractionInputInteractionView
+      assertThat(textAnswerRetriever.getPendingAnswer()).isInstanceOf(InteractionObject::class.java)
+      assertThat(textAnswerRetriever.getPendingAnswer().objectTypeCase).isEqualTo(InteractionObject.ObjectTypeCase.FRACTION)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.isNegative).isEqualTo(false)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.numerator).isEqualTo(9)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.denominator).isEqualTo(10)
+    }
+  }
+
+  @Test
+  fun testFractionInputInteractionView_withInputtedNegativeValue_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_fraction_input_interaction_view)).perform(typeText("-9/10"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_fraction_input_interaction_view) as FractionInputInteractionView
+      assertThat(textAnswerRetriever.getPendingAnswer()).isInstanceOf(InteractionObject::class.java)
+      assertThat(textAnswerRetriever.getPendingAnswer().objectTypeCase).isEqualTo(InteractionObject.ObjectTypeCase.FRACTION)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.isNegative).isEqualTo(true)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.numerator).isEqualTo(9)
+      assertThat(textAnswerRetriever.getPendingAnswer().fraction.denominator).isEqualTo(10)
+    }
+  }
+  @Test
+  fun testFractionInputInteractionView_withInputtedText_onConfigurationChange_hasCorrectPendingAnswer() {
+    onView(withId(R.id.test_fraction_input_interaction_view)).perform(typeText("9/5"))
+    activityScenario.onActivity { activity ->
+      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    }
+    onView(withId(R.id.test_fraction_input_interaction_view)).check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+      .check(
+        ViewAssertions.matches(ViewMatchers.withText("9/5"))
+      )
```

Comment:
You can directly import `ViewAssertions.matches`, `ViewMatchers.isDisplayed`, `ViewAssertions.withText`

`import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withText`

This will simplify this line to :

`onView(withId(R.id.test_number_input_interaction_view)).check(matches(isDisplayed())).check(matches(withText("9")))`

Also, after doing this make sure you remove unnecessary imports.

Ditto for other test-cases

Reply:
resolved